Python/Product/Debugger/Debugger/LegacyDebuggerProtocol.cs

```diff
@@ -0,0 +1,593 @@
+// Python Tools for Visual Studio
+// Copyright(c) Microsoft Corporation
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the License); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.PythonTools.Ipc.Json;
+
+namespace Microsoft.PythonTools.Debugger {
+    // IMPORTANT:
+    // Names of all fields, commands and events must match the names in visualstudio_py_debugger.py
+    internal static class LegacyDebuggerProtocol {
+        public static readonly Dictionary<string, Type> RegisteredTypes = CollectCommands();
+
+        private static Dictionary<string, Type> CollectCommands() {
+            Dictionary<string, Type> all = new Dictionary<string, Type>();
+            foreach (var type in typeof(LegacyDebuggerProtocol).GetNestedTypes()) {
+                if (type.IsSubclassOf(typeof(Request))) {
+                    var command = type.GetField("Command");
+                    if (command != null) {
+                        all["request." + (string) command.GetRawConstantValue()] = type;
+                    }
+                }
+                else if (type.IsSubclassOf(typeof(Event))) {
+                    var name = type.GetField("Name");
+                    if (name != null) {
+                        all["event." + (string) name.GetRawConstantValue()] = type;
+                    }
+                }
+            }
+            return all;
+        }
+
+        //////////////////////////////////////////////////////////////////////
+        /// Requests
+        //////////////////////////////////////////////////////////////////////
+
+        public sealed class StepIntoRequest : GenericRequest {
+            public const string Command = "legacyStepInto";
+
+            public override string command => Command;
+
+            public long threadId;
```

Comment:
Does JSON handle longs okay? Would we be better to send these as strings (in case someone takes the JavaScript part seriously and casts them to doubles)?